addons/barcodes/models/barcode_rule.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+import re
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class BarcodeRule(models.Model):
+    _name = 'barcode.rule'
+    _order = 'sequence asc'
+
+    def _encoding_selection_list(self):
```

Comment:
If it is going to be static list then we can remove the function, if someone wants to add new key in selection then he can add through selection_add